requirements_wheels.txt

```diff
@@ -0,0 +1,6 @@
+cffi==1.17.1
```

Comment:
I get the point, however, it messes with the separate of concerns.

For me, this is just the wrong place to do this.

Reply:
Is it the location, i.e. should it be done in a separate repo, or the design itself?
It's basically similar to the wheel build in https://github.com/home-assistant/wheels-custom-integrations/ just for core build dependencies which is why I thought if `core` isn't the right location, this repo might be.

Reply:
> It's basically similar to the wheel build in [home-assistant/wheels-custom-integrations](https://github.com/home-assistant/wheels-custom-integrations/) just for core build

Exactly. The custom wheels are not for core. These are; thus should be part of core.

../Frenck

Reply:
Ah, then I misunderstood your initial comment on https://github.com/home-assistant/core/pull/126987#pullrequestreview-2378450421. Though you didn't like it in `core`.

I can go ahead and port this PR over to core. However, I don't fully agree that they are misplaced here. I'd consider these dependencies requirements for all wheel builds thus this repo seems to be perfect for it.

As example, consider `cython` for a moment. They recently started uploading `py3-none-any` wheels as well. Thus without the explicit `skip-binary` wheel build here, the wheel builder would install a slow and unoptimized version for some of the platforms.

`numpy` and `cffi` are also basic requirements which are installed by default for the wheel builder. `ninja` is closely related as it's a build system used by among other things pandas.

The only outliers are `mysqlclient` and `psycopg2`. Those wheels arguably should be build in `core`. However, since they are completely optional it's only handled in `home-assistant/docker` for the used api version, currently.

--
Another advantage to bundling it here is that we don't require a new release before building the wheels. I.e. updating to `cp314` next year will be a bit easier.

Reply:
>  Though you didn't like it in core.

Correct, that PR build unpinned wheels (version ranged).

> However, I don't fully agree that they are misplaced here. I'd consider these dependencies requirements for all wheel builds thus this repo seems to be perfect for it.

I don't. We use this in more places, including places that actually don't use this at all. This repository should focus on being a wheels builder, not building wheels for use in child repositories using this builder.

> Another advantage to bundling it here is that we don't require a new release before building the wheels. I.e. updating to cp314 next year will be a bit easier.

That is an upstream issue (the consumer), not an issue for an builder.